// angular-ng-sidenav/esm5/lib/components/ng-sidenav/ng-sidenav.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/ng-sidenav/ng-sidenav.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { sidenavAnimations } from '../../sidenav-animations';
var NgSidenavComponent = /** @class */ (function () {
    function NgSidenavComponent() {
        this.expanded = true; // Whether drawer is open or close depend on drawer state
        // Whether drawer is open or close depend on drawer state
        this.drawerState = 'expanded'; // Drawer state (expanded, collapsed, hidden)
        // Drawer state (expanded, collapsed, hidden)
        this.color = '#19222a';
        this.drawerWidth = '250px';
        this.drawerWidthCollapsed = '80px';
    }
    /**
     * @return {?}
     */
    NgSidenavComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    NgSidenavComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        // Drawer state change
        if (changes && changes.drawerState) {
            this._drawerState = changes.drawerState.currentValue;
        }
        // drawer open/close change
        if (changes && changes.expanded) {
            this.expanded = changes.expanded.currentValue;
            this.drawerState = this.expanded ? 'expanded' : this._drawerState;
        }
    };
    NgSidenavComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-sidenav',
                    template: "<div class=\"ng-sidenav\"\n     [style.backgroundColor]=\"color\"\n     [@drawerState]=\"{value:drawerState,\n     params:{drawerWidth:drawerWidth, drawerWidthCollapsed: drawerWidthCollapsed}}\">\n  <ng-content></ng-content>\n</div>\n",
                    animations: [sidenavAnimations.serDrawerState],
                    styles: [".ng-sidenav{display:flex;flex-direction:column;height:100%;width:100%;box-sizing:border-box}"]
                }] }
    ];
    /** @nocollapse */
    NgSidenavComponent.ctorParameters = function () { return []; };
    NgSidenavComponent.propDecorators = {
        expanded: [{ type: Input }],
        drawerState: [{ type: Input }],
        color: [{ type: Input }],
        drawerWidth: [{ type: Input }],
        drawerWidthCollapsed: [{ type: Input }]
    };
    return NgSidenavComponent;
}());
export { NgSidenavComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgSidenavComponent.prototype._drawerState;
    /** @type {?} */
    NgSidenavComponent.prototype.expanded;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerState;
    /** @type {?} */
    NgSidenavComponent.prototype.color;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerWidth;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerWidthCollapsed;
}
