// angular-ng-sidenav/__ivy_ngcc__/fesm2015/angular-ng-sidenav.js
import { Component, Input, Directive, TemplateRef, ViewContainerRef, NgModule } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidenav.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
const _c1 = function (a0, a1) { return { drawerWidth: a0, drawerWidthCollapsed: a1 }; };
const _c2 = function (a0, a1) { return { value: a0, params: a1 }; };
class SidenavComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
SidenavComponent.ɵfac = function SidenavComponent_Factory(t) { return new (t || SidenavComponent)(); };
SidenavComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidenavComponent, selectors: [["ng-sidenav-container"]], ngContentSelectors: _c0, decls: 1, vars: 0, template: function SidenavComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%] {\n      display: flex;\n      align-items: stretch;\n      position: fixed;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n    }"] });
/** @nocollapse */
SidenavComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidenavComponent, [{
        type: Component,
        args: [{
                selector: 'ng-sidenav-container',
                template: '<ng-content></ng-content>',
                styles: [`
    :host {
      display: flex;
      align-items: stretch;
      position: fixed;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
    }
  `]
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidenav-animations.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Animation that slides a drawer in and out.
 * @type {?}
 */
const sidenavAnimations = {
    serDrawerState: trigger('drawerState', [
        state('expanded', style({ width: '{{ drawerWidth }}' }), { params: { drawerWidth: '250px' } }),
        state('collapsed', style({ width: '{{ drawerWidthCollapsed }}' }), { params: { drawerWidthCollapsed: '80px' } }),
        state('hidden', style({ width: '0' })),
        transition('collapsed => expanded', animate('200ms ease-in')),
        transition('expanded => collapsed', animate('200ms ease-out')),
        transition('hidden => expanded', animate('200ms ease-in')),
        transition('expanded => hidden', animate('200ms ease-out'))
    ]),
};

/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/ng-sidenav/ng-sidenav.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgSidenavComponent {
    constructor() {
        this.expanded = true; // Whether drawer is open or close depend on drawer state
        // Whether drawer is open or close depend on drawer state
        this.drawerState = 'expanded'; // Drawer state (expanded, collapsed, hidden)
        // Drawer state (expanded, collapsed, hidden)
        this.color = '#19222a';
        this.drawerWidth = '250px';
        this.drawerWidthCollapsed = '80px';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        // Drawer state change
        if (changes && changes.drawerState) {
            this._drawerState = changes.drawerState.currentValue;
        }
        // drawer open/close change
        if (changes && changes.expanded) {
            this.expanded = changes.expanded.currentValue;
            this.drawerState = this.expanded ? 'expanded' : this._drawerState;
        }
    }
}
NgSidenavComponent.ɵfac = function NgSidenavComponent_Factory(t) { return new (t || NgSidenavComponent)(); };
NgSidenavComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgSidenavComponent, selectors: [["ng-sidenav"]], inputs: { expanded: "expanded", drawerState: "drawerState", color: "color", drawerWidth: "drawerWidth", drawerWidthCollapsed: "drawerWidthCollapsed" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c0, decls: 2, vars: 9, consts: [[1, "ng-sidenav"]], template: function NgSidenavComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵproperty("@drawerState", ɵngcc0.ɵɵpureFunction2(6, _c2, ctx.drawerState, ɵngcc0.ɵɵpureFunction2(3, _c1, ctx.drawerWidth, ctx.drawerWidthCollapsed)));
    } }, styles: [".ng-sidenav[_ngcontent-%COMP%]{display:flex;flex-direction:column;height:100%;width:100%;box-sizing:border-box}"], data: { animation: [sidenavAnimations.serDrawerState] } });
/** @nocollapse */
NgSidenavComponent.ctorParameters = () => [];
NgSidenavComponent.propDecorators = {
    expanded: [{ type: Input }],
    drawerState: [{ type: Input }],
    color: [{ type: Input }],
    drawerWidth: [{ type: Input }],
    drawerWidthCollapsed: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSidenavComponent, [{
        type: Component,
        args: [{
                selector: 'ng-sidenav',
                template: "<div class=\"ng-sidenav\"\n     [style.backgroundColor]=\"color\"\n     [@drawerState]=\"{value:drawerState,\n     params:{drawerWidth:drawerWidth, drawerWidthCollapsed: drawerWidthCollapsed}}\">\n  <ng-content></ng-content>\n</div>\n",
                animations: [sidenavAnimations.serDrawerState],
                styles: [".ng-sidenav{display:flex;flex-direction:column;height:100%;width:100%;box-sizing:border-box}"]
            }]
    }], function () { return []; }, { expanded: [{
            type: Input
        }], drawerState: [{
            type: Input
        }], color: [{
            type: Input
        }], drawerWidth: [{
            type: Input
        }], drawerWidthCollapsed: [{
            type: Input
        }] }); })();
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgSidenavComponent.prototype._drawerState;
    /** @type {?} */
    NgSidenavComponent.prototype.expanded;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerState;
    /** @type {?} */
    NgSidenavComponent.prototype.color;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerWidth;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerWidthCollapsed;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/ng-sidenav-content/ng-sidenav-content.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgSidenavContentComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NgSidenavContentComponent.ɵfac = function NgSidenavContentComponent_Factory(t) { return new (t || NgSidenavContentComponent)(); };
NgSidenavContentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgSidenavContentComponent, selectors: [["ng-sidenav-content"]], ngContentSelectors: _c0, decls: 1, vars: 0, template: function NgSidenavContentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{flex:1;height:100%;overflow-y:auto;padding:20px;box-sizing:border-box}"] });
/** @nocollapse */
NgSidenavContentComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSidenavContentComponent, [{
        type: Component,
        args: [{
                selector: 'ng-sidenav-content',
                template: "<ng-content></ng-content>\n",
                styles: [":host{flex:1;height:100%;overflow-y:auto;padding:20px;box-sizing:border-box}"]
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * Generated from: lib/hidden-on-collapsed.directive.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class HiddenOnCollapsedDirective {
    /**
     * @param {?} templateRef
     * @param {?} viewContainer
     */
    constructor(templateRef, viewContainer) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    /**
     * @param {?} isShown
     * @return {?}
     */
    set hiddenOnCollapsed(isShown) {
        if (isShown) {
            // If condition is true add template to DOM
            this.viewContainer.createEmbeddedView(this.templateRef);
        }
        else {
            // Else remove template from DOM
            this.viewContainer.clear();
        }
    }
}
HiddenOnCollapsedDirective.ɵfac = function HiddenOnCollapsedDirective_Factory(t) { return new (t || HiddenOnCollapsedDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
HiddenOnCollapsedDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: HiddenOnCollapsedDirective, selectors: [["", "hiddenOnCollapsed", ""]], inputs: { hiddenOnCollapsed: "hiddenOnCollapsed" } });
/** @nocollapse */
HiddenOnCollapsedDirective.ctorParameters = () => [
    { type: TemplateRef },
    { type: ViewContainerRef }
];
HiddenOnCollapsedDirective.propDecorators = {
    hiddenOnCollapsed: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HiddenOnCollapsedDirective, [{
        type: Directive,
        args: [{
                selector: '[hiddenOnCollapsed]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }, { type: ɵngcc0.ViewContainerRef }]; }, { hiddenOnCollapsed: [{
            type: Input
        }] }); })();
if (false) {
    /**
     * @type {?}
     * @private
     */
    HiddenOnCollapsedDirective.prototype.templateRef;
    /**
     * @type {?}
     * @private
     */
    HiddenOnCollapsedDirective.prototype.viewContainer;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidenav.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class SidenavModule {
}
SidenavModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SidenavModule });
SidenavModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SidenavModule_Factory(t) { return new (t || SidenavModule)(); }, imports: [[CommonModule, BrowserAnimationsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SidenavModule, { declarations: function () { return [SidenavComponent,
        NgSidenavComponent,
        NgSidenavContentComponent,
        HiddenOnCollapsedDirective]; }, imports: function () { return [CommonModule, BrowserAnimationsModule]; }, exports: function () { return [SidenavComponent,
        NgSidenavComponent,
        NgSidenavContentComponent,
        HiddenOnCollapsedDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidenavModule, [{
        type: NgModule,
        args: [{
                declarations: [SidenavComponent, NgSidenavComponent, NgSidenavContentComponent, HiddenOnCollapsedDirective],
                imports: [CommonModule, BrowserAnimationsModule],
                exports: [SidenavComponent, NgSidenavComponent, NgSidenavContentComponent, HiddenOnCollapsedDirective]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * Generated from: public-api.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: angular-ng-sidenav.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { HiddenOnCollapsedDirective, NgSidenavComponent, NgSidenavContentComponent, SidenavComponent, SidenavModule, sidenavAnimations };
