// angular-ng-sidenav/fesm5/angular-ng-sidenav.js
import { Component, Input, Directive, TemplateRef, ViewContainerRef, NgModule } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidenav.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SidenavComponent = /** @class */ (function () {
    function SidenavComponent() {
    }
    /**
     * @return {?}
     */
    SidenavComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    SidenavComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-sidenav-container',
                    template: '<ng-content></ng-content>',
                    styles: ["\n    :host {\n      display: flex;\n      align-items: stretch;\n      position: fixed;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n    }\n  "]
                }] }
    ];
    /** @nocollapse */
    SidenavComponent.ctorParameters = function () { return []; };
    return SidenavComponent;
}());

/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidenav-animations.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Animation that slides a drawer in and out.
 * @type {?}
 */
var sidenavAnimations = {
    serDrawerState: trigger('drawerState', [
        state('expanded', style({ width: '{{ drawerWidth }}' }), { params: { drawerWidth: '250px' } }),
        state('collapsed', style({ width: '{{ drawerWidthCollapsed }}' }), { params: { drawerWidthCollapsed: '80px' } }),
        state('hidden', style({ width: '0' })),
        transition('collapsed => expanded', animate('200ms ease-in')),
        transition('expanded => collapsed', animate('200ms ease-out')),
        transition('hidden => expanded', animate('200ms ease-in')),
        transition('expanded => hidden', animate('200ms ease-out'))
    ]),
};

/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/ng-sidenav/ng-sidenav.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgSidenavComponent = /** @class */ (function () {
    function NgSidenavComponent() {
        this.expanded = true; // Whether drawer is open or close depend on drawer state
        // Whether drawer is open or close depend on drawer state
        this.drawerState = 'expanded'; // Drawer state (expanded, collapsed, hidden)
        // Drawer state (expanded, collapsed, hidden)
        this.color = '#19222a';
        this.drawerWidth = '250px';
        this.drawerWidthCollapsed = '80px';
    }
    /**
     * @return {?}
     */
    NgSidenavComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    NgSidenavComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        // Drawer state change
        if (changes && changes.drawerState) {
            this._drawerState = changes.drawerState.currentValue;
        }
        // drawer open/close change
        if (changes && changes.expanded) {
            this.expanded = changes.expanded.currentValue;
            this.drawerState = this.expanded ? 'expanded' : this._drawerState;
        }
    };
    NgSidenavComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-sidenav',
                    template: "<div class=\"ng-sidenav\"\n     [style.backgroundColor]=\"color\"\n     [@drawerState]=\"{value:drawerState,\n     params:{drawerWidth:drawerWidth, drawerWidthCollapsed: drawerWidthCollapsed}}\">\n  <ng-content></ng-content>\n</div>\n",
                    animations: [sidenavAnimations.serDrawerState],
                    styles: [".ng-sidenav{display:flex;flex-direction:column;height:100%;width:100%;box-sizing:border-box}"]
                }] }
    ];
    /** @nocollapse */
    NgSidenavComponent.ctorParameters = function () { return []; };
    NgSidenavComponent.propDecorators = {
        expanded: [{ type: Input }],
        drawerState: [{ type: Input }],
        color: [{ type: Input }],
        drawerWidth: [{ type: Input }],
        drawerWidthCollapsed: [{ type: Input }]
    };
    return NgSidenavComponent;
}());
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgSidenavComponent.prototype._drawerState;
    /** @type {?} */
    NgSidenavComponent.prototype.expanded;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerState;
    /** @type {?} */
    NgSidenavComponent.prototype.color;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerWidth;
    /** @type {?} */
    NgSidenavComponent.prototype.drawerWidthCollapsed;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/ng-sidenav-content/ng-sidenav-content.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgSidenavContentComponent = /** @class */ (function () {
    function NgSidenavContentComponent() {
    }
    /**
     * @return {?}
     */
    NgSidenavContentComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    NgSidenavContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-sidenav-content',
                    template: "<ng-content></ng-content>\n",
                    styles: [":host{flex:1;height:100%;overflow-y:auto;padding:20px;box-sizing:border-box}"]
                }] }
    ];
    /** @nocollapse */
    NgSidenavContentComponent.ctorParameters = function () { return []; };
    return NgSidenavContentComponent;
}());

/**
 * @fileoverview added by tsickle
 * Generated from: lib/hidden-on-collapsed.directive.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var HiddenOnCollapsedDirective = /** @class */ (function () {
    function HiddenOnCollapsedDirective(templateRef, viewContainer) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    Object.defineProperty(HiddenOnCollapsedDirective.prototype, "hiddenOnCollapsed", {
        set: /**
         * @param {?} isShown
         * @return {?}
         */
        function (isShown) {
            if (isShown) {
                // If condition is true add template to DOM
                this.viewContainer.createEmbeddedView(this.templateRef);
            }
            else {
                // Else remove template from DOM
                this.viewContainer.clear();
            }
        },
        enumerable: true,
        configurable: true
    });
    HiddenOnCollapsedDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[hiddenOnCollapsed]'
                },] }
    ];
    /** @nocollapse */
    HiddenOnCollapsedDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef }
    ]; };
    HiddenOnCollapsedDirective.propDecorators = {
        hiddenOnCollapsed: [{ type: Input }]
    };
    return HiddenOnCollapsedDirective;
}());
if (false) {
    /**
     * @type {?}
     * @private
     */
    HiddenOnCollapsedDirective.prototype.templateRef;
    /**
     * @type {?}
     * @private
     */
    HiddenOnCollapsedDirective.prototype.viewContainer;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidenav.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SidenavModule = /** @class */ (function () {
    function SidenavModule() {
    }
    SidenavModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [SidenavComponent, NgSidenavComponent, NgSidenavContentComponent, HiddenOnCollapsedDirective],
                    imports: [CommonModule, BrowserAnimationsModule],
                    exports: [SidenavComponent, NgSidenavComponent, NgSidenavContentComponent, HiddenOnCollapsedDirective]
                },] }
    ];
    return SidenavModule;
}());

/**
 * @fileoverview added by tsickle
 * Generated from: public-api.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: angular-ng-sidenav.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { HiddenOnCollapsedDirective, NgSidenavComponent, NgSidenavContentComponent, SidenavComponent, SidenavModule, sidenavAnimations };
